source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -73,6 +74,25 @@ std::vector<Http::HeaderUtility::HeaderData> headerMatchers(const T& matcher_pro
   return matchers;
 }
 
+// Transforms the proto list of 'auth_scopes' into a vector of std::string, also
+// handling the default value logic
+template <class T> std::vector<std::string> authScopesList(const T& auth_scopes_protos) {
+  std::vector<std::string> scopes;
+
+  // if 'auth_scopes' is empty it must return a list with the default value.
+  if (auth_scopes_protos.empty()) {
+    scopes.reserve(1);
```

Comment:
Reserving one seems redundant, I imagine it's only useful to avoid multiple allocations from happening by telling the vector that multiple entries will be inserted

Reply:
@snowp do you think just removing `scopes.reserve(1);` would be enough here?

Reply:
yea

Reply:
Removed.